gelataio_boy_control/src/scooping.cpp

```diff
@@ -171,9 +172,58 @@ bool ScoopingMain::perform_scoop() {
     return right_arm.moveJoint("wrist_right", -0.5);
 }
 
-bool ScoopingMain::depart_from_scoop() {
-    right_arm.setPlanningTime(5.0);
-    ROS_WARN("Depart from scoop motion not yet implemented");
-    return true;
+bool ScoopingMain::drop_ice() {
+    ROS_WARN("Rotating the wrist_right");
+
+    return right_arm.moveJoint("wrist_right", 0.87);
 }
 
+bool ScoopingMain::depart_from_scoop(geometry_msgs::Point point_above_cup) {
+    right_arm.setPlanningTime(15.0);
+    Pose hold_scoop_pose;
+    hold_scoop_pose.position = point_above_cup;
+    moveit_msgs::JointConstraint wristConstraint;
+    moveit_msgs::Constraints constraints;
+    tf2::Quaternion q_start;
+    double roll = -20;
+    double pitch = 40;
+    double yaw = 45 + 90;
+    bool scooper_hold = false;
+    bool drop_success = false;
+
+    q_start.setRPY(roll/180*M_PI, pitch/180*M_PI, yaw/180*M_PI);
+    hold_scoop_pose.orientation.x = q_start.x();
+    hold_scoop_pose.orientation.y = q_start.y();
+    hold_scoop_pose.orientation.z = q_start.z();
+    hold_scoop_pose.orientation.w = q_start.w();
+
+    wristConstraint.joint_name = "wrist_right";
+    wristConstraint.position = 0.01;
+    wristConstraint.tolerance_below = .1;
+    wristConstraint.tolerance_above = .1;
+    wristConstraint.weight = 1.0;
+
+    constraints.joint_constraints.push_back(wristConstraint);
+    
+    ROS_ERROR("Before departing");
+
+    // Wrist right go to zero and then it should nevere move again
+    // We should add the box as a constrain
+    scooper_hold = right_arm.moveToPose(hold_scoop_pose, constraints);
```

Comment:
moveToPose returns true if planning was successful, false otherwise. I don't get the logic here. So why could the scooper spill icecream if he couldn't find a plan?

Reply:
fixed